SimPPS/PPSPixelDigiProducer/plugins/PPSPixelDigiAnalyzer.cc

```diff
@@ -0,0 +1,309 @@
+#ifndef SimPPS_RPIXDigiAnalyzer_h
+#define SimPPS_RPIXDigiAnalyzer_h
+
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+#include <FWCore/Framework/interface/EDAnalyzer.h>
+#include <DataFormats/CTPPSDetId/interface/CTPPSPixelDetId.h>
+#include "SimDataFormats/TrackingHit/interface/PSimHitContainer.h"
+#include "SimDataFormats/TrackingHit/interface/PSimHit.h"
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigi.h"
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigiCollection.h"
+#include "Geometry/VeryForwardGeometry/interface/CTPPSPixelSimTopology.h"
+
+#include "SimDataFormats/CrossingFrame/interface/CrossingFrame.h"
+#include "SimDataFormats/CrossingFrame/interface/MixCollection.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include <FWCore/Framework/interface/Event.h>
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/GeometryVector/interface/LocalPoint.h"
+
+#include <iostream>
+#include <string>
+
+#include "TFile.h"
+#include "TH2D.h"
+
+#define SELECTED_PIXEL_ROW 89
+#define SELECTED_PIXEL_COLUMN 23
+#define SELECTED_UNITID 2014314496
+#define TG184 0.332655724
+
+#define USE_MIDDLE_OF_PIXEL_2
+#define CENTERX 1.05
+#define CENTERY -8.475
+
+using namespace edm;
+using namespace std;
+
+class TFile;
+class PSimHit;
+
+namespace edm {
+  class ParameterSet;
+  class Event;
+  class EventSetup;
+}  // namespace edm
+
+class PPSPixelDigiAnalyzer : public edm::EDAnalyzer {
```

Comment:
@mundim , in this analizer you open root files. In that case, better to inherit of "one" type of analyzers

class PPSPixelDigiAnalyze : public edm::one::EDAnalyzer<edm::one::SharedResources> {

Reply:
done